src/main/java/com/troubleshooting/troubleshootingtool/ContainerDataRetrieval.java

```diff
@@ -0,0 +1,69 @@
+package com.troubleshooting.troubleshootingtool;
+
+import com.troubleshooting.model.ConjurLogModel;
+import com.troubleshooting.model.EnvModel;
+
+import com.troubleshooting.model.EnvsModel;
+import com.troubleshooting.model.LogsModel;
+import org.springframework.stereotype.Component;
+
+import java.util.List;
+
+// Retrieves the data returned from the Docker Client and packs the data in the appropriate data models
+@Component
+public class ContainerDataRetrieval implements IDataRetrieval {
+    ContainerDataAccess access = new ContainerDataAccess();
+
+    ConjurLogModel log = new ConjurLogModel();
+    LogsModel logs = new LogsModel();
+
+    EnvModel env = new EnvModel();
+    EnvsModel envs = new EnvsModel();
+
+    //	Extract Logs
+    @Override
+    public LogsModel getData(String containerID) {
+        List<String> loggingFrames = access.getLogs(containerID);
+
+        for(int i = 0; i < loggingFrames.size(); i++) {
+            System.out.println(loggingFrames.get(i));
+            logParser(loggingFrames.get(i), log);
+        }
+        return logs;
+    }
+
+    // Extracts important parts of each Log entry
+    public void logParser(String logInstance, ConjurLogModel log) {
+        String[] logEntry = logInstance.split(" ", 8);
+
+        if (!logEntry[0].contains("STDERR:")) {
```

Comment:
- Check the string for null
- Check the array boundaries
- Use a constant for "STDERR:"